pkg/component/autoscaling/vpa/general.go

```diff
@@ -205,12 +206,24 @@ func (v *vpa) reconcileGeneralMutatingWebhookConfiguration(mutatingWebhookConfig
 			"intentionally starts with 'zzz', such that it is called after all other webhooks which inject "+
 			"containers. All containers injected by webhooks that are called _after_ the vpa webhook will not be "+
 			"under control of vpa.")
+
+	if v.values.ClusterType == component.ClusterTypeSeed {
+		namespaceSelector = &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{Key: "name", Operator: metav1.LabelSelectorOpNotIn, Values: []string{"kube-system"}},
+			},
+		}
+	} else {
+		namespaceSelector = nil
+	}
```

Comment:
Nit: IMO, no need of this else brach as it is anyways nil by default.

Reply:
Done - removed.